pkg/path/parser/parser_test.go

```diff
@@ -0,0 +1,245 @@
+			},
+		},
+		{
+			input: `spec.containers[name: *].securityContext`,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+				&List{KeyField: "name", Glob: true},
+				&Object{Value: "securityContext"},
+			},
+		},
+		{
+			// A quoted '*' is a fieldValue, not a glob
+			input: `spec.containers[name: "*"].securityContext`,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+				&List{KeyField: "name", KeyValue: strPtr("*"), Glob: false},
+				&Object{Value: "securityContext"},
+			},
+		},
+		{
+			input: `spec.containers[name: foo].securityContext`,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+				&List{KeyField: "name", KeyValue: strPtr("foo")},
+				&Object{Value: "securityContext"},
+			},
+		},
+		{
+			input: `spec.containers["my key": "foo bar"]`,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+				&List{KeyField: "my key", KeyValue: strPtr("foo bar")},
+			},
+		},
+		{
+			// Error: keys with whitespace must be quoted
+			input:     `spec.containers[my key: "foo bar"]`,
+			expectErr: true,
+		},
+		{
+			// Error: values with whitespace must be quoted
+			input:     `spec.containers[key: foo bar]`,
+			expectErr: true,
+		},
+		{
+			input: `spec.containers[name: ""].securityContext`,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+				&List{KeyField: "name", KeyValue: strPtr("")},
+				&Object{Value: "securityContext"},
+			},
+		},
+		{
+			// TODO: Is this useful or should we make it a parsing error?
+			input: `spec.containers[name: ].securityContext`,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+				&List{KeyField: "name", KeyValue: nil},
+				&Object{Value: "securityContext"},
+			},
+		},
+		{
+			// parse error: listSpec requires keyField
+			input:     `spec.containers[].securityContext`,
+			expectErr: true,
+		},
+		{
+			input:     `spec.containers[:].securityContext`,
+			expectErr: true,
+		},
+		{
+			input:     `spec.containers[:foo].securityContext`,
+			expectErr: true,
+		},
+		{
+			input:     `spec.containers[foo].securityContext`,
+			expectErr: true,
+		},
+		{
+			// parse error: we don't allow empty segments
+			input:     `foo..bar`,
+			expectErr: true,
+		},
+		{
+			// ...but we do allow zero-string-named segments
+			input: `foo."".bar`,
+			expected: []Node{
+				&Object{Value: "foo"},
+				&Object{Value: ""},
+				&Object{Value: "bar"},
+			},
+		},
+		{
+			// whitespace can surround tokens
+			input: `    spec   .    containers    `,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+			},
+		},
+		{
+			// whitespace can surround tokens
+			input: `    spec   .    "containers"    `,
+			expected: []Node{
+				&Object{Value: "spec"},
+				&Object{Value: "containers"},
+			},
+		},
+		{
+			// TODO: Should this be allowed?
```

Comment:
It should not be allowed. It's impossible to have a list-of-lists, as an intermediate object is required to store the key fields:

```
{
   "spec": [
      {
           "myKey": "some-value",
           "nestedList": [
                 {"secondKey": "secondValue"}
           ]
}
```

Note the need to access `nestedList`